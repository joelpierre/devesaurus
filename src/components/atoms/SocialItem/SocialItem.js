import React from 'react';
import PropTypes from 'prop-types';

const SocialItem = (props) => {
  return (<></>);
};

SocialItem.defaultProps = {};

SocialItem.propTypes = {};

export default SocialItem;
