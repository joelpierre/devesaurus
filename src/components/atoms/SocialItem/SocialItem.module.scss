@import "~sass/core";

.social-item {
  $root: &;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  border-radius: 50%;
  margin-right: 2rem;
  overflow: hidden;
  border: 2px solid $color-alpha;
  background: $color-gradient-brand;
  background: $color-alpha;
  transition: background .2s ease-in-out;

  &:last-of-type {
    margin-right: 0;
  }

  &__link {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;

    & > svg {
      width: auto !important;
      height: 50% !important;
      color: $color-tint-alpha;
      transition: color .2s ease-in-out;
    }
  }

  @include hover-focus {
    background: $color-tint-alpha;

    #{$root} {
      &__link {
        & > svg {
          color: $color-alpha;
        }
      }
    }
  }
}
