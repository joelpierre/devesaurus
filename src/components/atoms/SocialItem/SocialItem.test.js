import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

import SocialItem from './SocialItem';

const defaultProps = {
  name: 'test-name',
  link: 'test-link',
};

/**
 * Factory function to create a ShallowWrapper for the SocialItem component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<SocialItem {...setupProps} />);
};

describe('<SocialItem/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the SocialItem Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-social-item');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(SocialItem, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
