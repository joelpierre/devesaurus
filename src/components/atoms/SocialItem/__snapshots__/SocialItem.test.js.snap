// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialItem/> Should render SocialItem Component 1`] = `ShallowWrapper {}`;
